aiohttp_sse/__init__.py

```diff
@@ -149,49 +166,86 @@ def last_event_id(self) -> Optional[str]:
 
         return self._req.headers.get(self.DEFAULT_LAST_EVENT_HEADER)
 
+    @property
+    def ping_interval(self) -> Union[int, float]:
+        """Time interval between two ping massages"""
+        return self._ping_interval
+
     @ping_interval.setter
-    def ping_interval(self, value):
+    def ping_interval(self, value: Union[int, float]) -> None:
         """Setter for ping_interval property.
 
         :param int value: interval in sec between two ping values.
         """
 
-        if not isinstance(value, int):
-            raise TypeError("ping interval must be int")
+        if not isinstance(value, (int, float)):
+            raise TypeError("ping interval must be int or float")
         if value < 0:
             raise ValueError("ping interval must be greater then 0")
 
         self._ping_interval = value
 
-    async def _ping(self):
+    async def _ping(self) -> None:
         # periodically send ping to the browser. Any message that
         # starts with ":" colon ignored by a browser and could be used
         # as ping message.
         while True:
             await asyncio.sleep(self._ping_interval)
             try:
                 await self.write(": ping{0}{0}".format(self._sep).encode("utf-8"))
-            except ConnectionResetError:
-                self._ping_task.cancel()
+            except ConnectionResetError as exc:
+                raise asyncio.CancelledError() from exc
```

Comment:
What's this? This seems like totally different behaviour, and shouldn't be in a typing PR.

Reply:
My investigation history:

## 0. Original code

Mypy said than since `self._ping_task` is `Optional`, we can't call `.cancel` from `None`

## 1. Added check for mypy

```suggestion
                if self._ping_task is not None:
                    self._ping_task.cancel()
```

This way `mypy` is happy, but `coverage` is not! It said that there's no case when `self._ping_task is None` in that function. Cause this coro task saved in var `self._ping_task`.

## 2. Added direct cancel

Latest version - is what you see. Ping task cancels self inside. I.e. makes his "done"
Also alternative:

```suggestion
                    task = asyncio.current_task()
                    task.cancel()
```

Reply:
OK, I only just realised it is cancelling itself, which is a rather odd approach.
I think the most sensible option would be to debug log the exception and then `return` or `break`.
But, can we do this in a separate PR as well?